mllib-dal/src/main/native/CorrelationOneAPIImpl.cpp

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * Copyright 2020 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include <chrono>
+
+#ifdef CPU_GPU_PROFILE
+#include "GPU.h"
+#ifndef ONEDAL_DATA_PARALLEL
+#define ONEDAL_DATA_PARALLEL
+#endif
+
+#include "Communicator.hpp"
+#include "OutputHelpers.hpp"
+#include "com_intel_oap_mllib_stat_CorrelationDALImpl.h"
+#include "oneapi/dal/algo/covariance.hpp"
+#include "oneapi/dal/table/homogen.hpp"
+#include "service.h"
+
+using namespace std;
+using namespace oneapi::dal;
+const int ccl_root = 0;
+
+static void doCorrelationOneAPICompute(JNIEnv *env, jint rankId,
+                                       jlong pNumTabData, jint executorNum,
+                                       const ccl::string &ipPort,
+                                       jint computeDeviceOrdinal,
+                                       jobject resultObj) {
+    std::cout << "oneDAL (native): GPU/CPU compute start , rankid = %ld "
```

Comment:
could you just print "GPU compute start" or "CPU compute start"  and rankid instead of lengthy output?
Pls don't use %ld in this std::cout style output.